### PR TITLE
fix: [PL-27049] Adding open telemetry changes for platform and access-control

### PR DESCRIPTION
**Pull request by kshitij-harness**
## Describe your changes
Added Opentelemetry agent to platform service and access-control service

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35979)

### CODE REVIEW

File: access-control/build/container/scripts/start_process.sh [hunk 1]
```diff
@@ -36,6 +36,16 @@ if [[ "${ENABLE_APPDYNAMICS}" == "true" ]]; then
     echo "Using Appdynamics java agent"
 fi
 
+if [[ "${ENABLE_OPENTELEMETRY}" == "true" ]] ; then
+    echo "OpenTelemetry is enabled"
+    JAVA_OPTS=$JAVA_OPTS" -javaagent:/opt/harness/opentelemetry-javaagent.jar -Dotel.service.name=${OTEL_SERVICE_NAME:-pipeline-service}"
```
Comment (ashishsanodia-harness) [inline access-control/build/container/scripts/start_process.sh:41]: here default service name is "pipeline-service" which is not correct.

**Review by kshitij-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35979)* status: 2 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/build/container/scripts/start_process.sh` line 41 at r1](https://reviewable.io/reviews/harness/harness-core/35979#-N8YA4WcBfbrJ6bleNqd-r1-41:-N8YOWmHBymXIGuh7wa1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/69784e97453b85c5ca0723f4d63a73b3fd499336/access-control/build/container/scripts/start_process.sh#L41)):*
<details><summary><i>Previously, ashishsanodia-harness (Ashish Sanodia) wrote…</i></summary><blockquote>

here default service name is "pipeline-service" which is not correct.
</blockquote></details>

Done.
___
*[`platform-service/build/container/scripts/start_process.sh` line 45 at r1](https://reviewable.io/reviews/harness/harness-core/35979#-N8YA4WcBfbrJ6bleNqe-r1-45:-N8YOXn-8PrTVa7D6D7t:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/69784e97453b85c5ca0723f4d63a73b3fd499336/platform-service/build/container/scripts/start_process.sh#L45)):*
<details><summary><i>Previously, ashishsanodia-harness (Ashish Sanodia) wrote…</i></summary><blockquote>

here default service name is "pipeline-service" which is not correct.
</blockquote></details>

Done.

**Review by reetika04 — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35979)* status: 2 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by piyush-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35979)* status: 2. unresolved discussions.

**Review by ashishsanodia-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35979)* status: :shipit: complete! ✓

**Pull request merged**